src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -690,7 +691,7 @@ public void getTRCFModel(String modelId, ActionListener<Optional<ThresholdedRand
                         response -> deserializeTRCFModel(response, modelId, listener),
                         exception -> {
                             // expected exception, don't print stack trace
-                            if (exception instanceof IndexNotFoundException) {
+                            if (exception instanceof OpenSearchStatusException) {
```

Comment:
We should check for the index not found message rather directly checking for `OpenSearchStatusException` here?

Reply:
sure ill make this change